st2common/st2common/services/triggers.py

```diff
@@ -46,6 +49,27 @@ def get_trigger_db_given_type_and_params(type=None, parameters=None):
 
         trigger_db = trigger_dbs[0] if len(trigger_dbs) > 0 else None
 
+        # Work around for cron-timer when in some scenarios finding an object fails when Python
+        # value types are unicode :/
+        if not trigger_db and six.PY2 and type == CRON_TIMER_TRIGGER_REF and parameters:
+            non_unicode_literal_parameters = {}
```

Comment:
`.encode()` returns str in Python 2.x and bytes in 3.x. `.decode()` on string returns unicode in 3.x (in short, it's vice-versa as commonly though).